core/trino-main/src/main/java/io/trino/cost/TableScanStatsRule.java

```diff
@@ -59,7 +60,7 @@ public TableScanStatsRule(Metadata metadata, StatsNormalizer normalizer)
     protected Optional<PlanNodeStatsEstimate> doCalculate(TableScanNode node, StatsProvider sourceStats, Lookup lookup, Session session, TypeProvider types)
     {
         // TODO Construct predicate like AddExchanges's LayoutConstraintEvaluator
-        Constraint constraint = new Constraint(metadata.getTableProperties(session, node.getTable()).getPredicate());
+        Constraint constraint = new Constraint(TupleDomain.all());
```

Comment:
Do you plan to add stats rule for filter + table scan so that constraint param is actually used?

Reply:
Definitely out of  scope of this PR and i actually doubt it's needed. 
We rely on the fact applyFilter() is called prior reading stats (where it matters) and it seem to serve us wel